docs/dynamic-bundling/index.md

```diff
@@ -40,3 +40,130 @@ Most others (YUI, `ResourceLoader`) focus more on pure concatenation without exp
+          the conditional chunks are somehow hidden from direct requests.
+
+### Simplified Deployment Pipeline
+
+*Applies to: `begin.com`, `ResourceLoader`.*
+
+This use case is very different from the others and it's no coincident that it
+led to very different solutions. The underlying idea is still the same: Instead
+of generating the final response body ahead of time and deploying static files,
+the final assets are generated dynamically after the deploy.
+
+The basic requirement is: Source files can be pushed to production without any
+additional steps and can be requested almost immediately. Any transformation
+like concat/rollup of files or minification needs to be fast enough to happen
+during request handling. Generally the result can still be served from a CDN
+but on a cache miss, e.g. immediately following a deploy, bundling happens live.
+
+In this use case, dynamic bundling is used to reduce the lead time, specifically
+the time it takes to release changes.
+
+## (No) Client Cache Manifest
+
+There's an inheritent tension between the desire to batch resources and cache
+stability. On one hand combining resources is good since it reduces roundtrips
+and allows efficient compression. Even in the era of http2, there is a cost to
+making more requests. But batched resources are less cachable. There's no easy
+way to isolate cache invalidation to just a small unit when it's always included
+in a combined response.
+
+The same kind of problem can be seen with different technologies:
+
+1. http2 push: How to only push resources that the client needs?
+2. Dynamic Bundling: How to only send chunks the client hasn't cached yet?
+3. Web Bundles: How to prune already cached resources from a bundle?
+4. ReactNative: How to only send deltas instead of full application builds when
+                doing over-the-air updates?
+
+In all cases, the communication channel for requesting the bundle doesn't
+generally have sufficient bandwidth to transfer a complete description of the
+client cache state - if it's even possible to capture it.
+
+So far a few strategies have been observed to deal with this problem:
+
+1. Bundle Version Diff: The client attaches the last version it is aware of.
+   It receives a diff containing the latest contents. This assumes that the
+   client caches the entire chunk set (at least as it applies to the user).
+   Usual solution for over-the-air updates in ReactNative (e.g. AppCenter).
+2. Push all: This is the approach used by http2, dynamic bundling, and web
+   bundles. All resources are pushed to the client even if the client already
+   knows about them, e.g. because it downloaded a previous version of the
+   bundle. The client may ignore the contents, in the case of http2 even cancel
+   the transfer, but the server will attempt to send all resources at first.
+
+For completeness sake, there's also a strategy for h2 push that uses a cookie
```

Comment:
nit. should we say `http2` instead of `h2`? For a second I thought you were talking about the `h2` element :D

Reply:
Yeah, better to be explicit (and consistent). Thanks!